src/main/java/net/sf/mzmine/modules/peaklistmethods/io/spectraldbsubmit/LibrarySubmitParameters.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2006-2018 The MZmine 2 Development Team
+ * 
+ * This file is part of MZmine 2.
+ * 
+ * MZmine 2 is free software; you can redistribute it and/or modify it under the terms of the GNU
+ * General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * MZmine 2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License along with MZmine 2; if not,
+ * write to the Free Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301
+ * USA
+ */
+
+/*
+ * This module was prepared by Abi Sarvepalli, Christopher Jensen, and Zheng Zhang at the Dorrestein
+ * Lab (University of California, San Diego).
+ * 
+ * It is freely available under the GNU GPL licence of MZmine2.
+ * 
+ * For any questions or concerns, please refer to:
+ * https://groups.google.com/forum/#!forum/molecular_networking_bug_reports
+ * 
+ * Credit to the Du-Lab development team for the initial commitment to the MGF export module.
+ */
+
+package net.sf.mzmine.modules.peaklistmethods.io.spectraldbsubmit;
+
+import java.text.DecimalFormat;
+import net.sf.mzmine.main.MZmineCore;
+import net.sf.mzmine.parameters.Parameter;
+import net.sf.mzmine.parameters.impl.SimpleParameterSet;
+import net.sf.mzmine.parameters.parametertypes.BooleanParameter;
+import net.sf.mzmine.parameters.parametertypes.ComboParameter;
+import net.sf.mzmine.parameters.parametertypes.DoubleParameter;
+import net.sf.mzmine.parameters.parametertypes.OptionalParameter;
+import net.sf.mzmine.parameters.parametertypes.PasswordParameter;
+import net.sf.mzmine.parameters.parametertypes.StringParameter;
+import net.sf.mzmine.parameters.parametertypes.filenames.FileNameParameter;
+
+/**
+ * 
+ * @author Robin Schmid (robinschmid@uni-muenster.de)
+ *
+ */
+public class LibrarySubmitParameters extends SimpleParameterSet {
+
+  /**
+   * TODO difference of COMPOUND MASS and EXACT mass
+   * 
+   * @author r_schm33
+   *
+   */
+
+  public enum CompoundSource {
+  Lysate, Isolated, Commercial, Crude, Other;
+  }
+
+  public enum Polarity {
+    Positive, Negative;
+  }
+
+  public enum IonSource {
+    LC_ESI("LC-ESI"), DI_ESI("DI-ESI"), EI, APCI, ESI;
+
+    private final String value;
+
+    private IonSource() {
+      this.value = null;
+    }
+
+    private IonSource(String value) {
+      this.value = value;
+    }
+
+    @Override
+    public String toString() {
+      return value != null ? value : super.toString();
+    }
+  }
+  public enum Instrument {
```

Comment:
I would suggest putting this enum into a separate file, for clarity

Reply:
All enums are now located in a separate class (GnpsValues) to link them to the specific subject
this allows GnpsValues.Instrument. ...

Is that ok?